ImageLounge/src/DkGui/DkNoMacs.cpp

```diff
@@ -578,44 +612,74 @@ void DkNoMacs::toggleFullScreen()
 
 void DkNoMacs::enterFullScreen()
 {
-    DkSettingsManager::param().app().currentAppMode += qFloor(DkSettings::mode_end * 0.5f);
-    if (DkSettingsManager::param().app().currentAppMode < 0) {
-        qDebug() << "illegal state: " << DkSettingsManager::param().app().currentAppMode;
-        DkSettingsManager::param().app().currentAppMode = DkSettings::mode_default;
-    }
+    int appMode = DkSettingsManager::param().app().currentAppMode;
+    appMode = DkSettings::fullscreenMode(appMode);
+    DkSettingsManager::param().app().currentAppMode = appMode;
 
     menuBar()->hide();
     DkToolBarManager::inst().show(false);
     DkStatusBarManager::instance().statusbar()->hide();
     getTabWidget()->showTabs(false);
-
     restoreDocks();
 
-    // here is an issue with windows that I can't quite fix:
-    // if we send nomacs to fullscreen from an attached window (i.e. split window)
+    qInfo() << "before enter fullscreen appMode:" << appMode << "geometry:" << geometry() << "normalGeometry:" << normalGeometry()
+            << "windowState:" << windowState();
+
+    // setWindowState is buggy, does it have any benefit?
+    // -maximized->fullscreen doesn't cover the screen (Qt 6.5.1 / Windows 10)
+    // -maximized->fullscreen->maximized->normal loses normal size (Qt 5.15.13 / Linux / GNOME)
+    // #define USE_WINDOWSTATE 1
+
+#ifdef USE_WINDOWSTATE
     setWindowState(windowState() | Qt::WindowFullScreen);
+#else
+    mWasMaximized = isMaximized();
+    showFullScreen();
+#endif
```

Comment:
According to [the source code of `showFullScreen()`](https://github.com/qt/qtbase/blob/6c036012b5f2304a05af29f29daa7582603f79ae/src/widgets/kernel/qwidget.cpp#L3038-L3049), it would seem that `Qt::WindowMaximized` and `Qt::WindowFullScreen` does not mix. We should use the higher level `showFullScreen()` and remove the `setWindowState` branch. This should be safe because I do not see `USE_WINDOWSTATE` defined anywhere else.

Reply:
Thanks for looking into this, I left it in for testing but there is no need.